src/linux/sentry_procmaps_modulefinder.c

```diff
@@ -232,14 +251,14 @@ load_modules(void)
         if (!read) {
             break;
         }
-        // skip over anonymous mappings
-        if (!module.file
-            || (module.file[0] == '[' && strcmp("[vdso]", module.file) != 0)) {
+        // skip over anonymous mappings, but keep the one for the linux vdso
+        if (module.start != linux_vdso && !module.file
```

Comment:
For the vdso image, you still need to overwrite the file name with `"linux-gate.so"`.